src/List.ts

```diff
@@ -30,7 +32,7 @@ export class List<Value> implements Category<Value, 'List'> {
         return this.values;
     }
     public concat(x: Value): List<Value> {
-        return new List(this.values.concat(x));
+        return new List([...this.values, x]);
```

Comment:
this is not essential but if you want to be efficient you should take care of not allocating memory twice: the first time to build `[...this.values, x]` and the second time to allocate the `new List`.
May be a constructor with two arguments (`values` and `x`) could avoid useless memory allocation.